timemachine/fe/single_topology_v3.py

```diff
@@ -0,0 +1,539 @@
+    mol_a_angle_params, mol_a_ha = mol_a_top.parameterize_harmonic_angle(ff.ha_handle.params)
+    mol_a_proper_params, mol_a_pt = mol_a_top.parameterize_proper_torsion(ff.pt_handle.params)
+    mol_a_improper_params, mol_a_it = mol_a_top.parameterize_improper_torsion(ff.it_handle.params)
+    mol_a_nbpl_params, mol_a_nbpl = mol_a_top.parameterize_nonbonded_pairlist(ff.q_handle.params, ff.lj_handle.params)
+
+    mol_a_bond_params = mol_a_bond_params.tolist()
+    mol_a_angle_params = mol_a_angle_params.tolist()
+    mol_a_proper_params = mol_a_proper_params.tolist()
+    mol_a_improper_params = mol_a_improper_params.tolist()
+    mol_a_nbpl_params = mol_a_nbpl_params.tolist()
+
+    mol_a_bond_idxs = recursive_map(mol_a_hb.get_idxs(), a_to_c)
+    mol_a_angle_idxs = recursive_map(mol_a_ha.get_idxs(), a_to_c)
+    mol_a_proper_idxs = recursive_map(mol_a_pt.get_idxs(), a_to_c)
+    mol_a_improper_idxs = recursive_map(mol_a_it.get_idxs(), a_to_c)
+    mol_a_nbpl_idxs = recursive_map(mol_a_nbpl.get_idxs(), a_to_c)
+
+    all_dummy_bond_idxs = recursive_map(all_dummy_bond_idxs, b_to_c)
+    all_dummy_angle_idxs = recursive_map(all_dummy_angle_idxs, b_to_c)
+    all_dummy_improper_idxs = recursive_map(all_dummy_improper_idxs, b_to_c)
+
+    # parameterize the combined molecule
+    mol_c_bond_idxs = mol_a_bond_idxs + all_dummy_bond_idxs
+    mol_c_bond_params = mol_a_bond_params + all_dummy_bond_params
+
+    mol_c_angle_idxs = mol_a_angle_idxs + all_dummy_angle_idxs
+    mol_c_angle_params = mol_a_angle_params + all_dummy_angle_params
+
+    mol_c_proper_idxs = mol_a_proper_idxs
+    mol_c_proper_params = mol_a_proper_params
+
+    mol_c_improper_idxs = mol_a_improper_idxs + all_dummy_improper_idxs
+    mol_c_improper_params = mol_a_improper_params + all_dummy_improper_params
+
+    # combine proper + improper
+    mol_c_torsion_idxs = mol_c_proper_idxs + mol_c_improper_idxs
+    mol_c_torsion_params = mol_c_proper_params + mol_c_improper_params
+
+    mol_c_nbpl_idxs = mol_a_nbpl_idxs
+    num_combined_atoms = mol_a.GetNumAtoms() + mol_b.GetNumAtoms() - len(core)
+    mol_c_nbpl_params = np.zeros((num_combined_atoms, 3))
+    for a_idx, a_param in enumerate(mol_a_nbpl_params):
+        mol_c_nbpl_params[a_to_c[a_idx]] = a_param
+
+    bond_potential = potentials.HarmonicBond(mol_c_bond_idxs).bind(mol_c_bond_params)
+    angle_potential = potentials.HarmonicAngle(mol_c_angle_idxs).bind(mol_c_angle_params)
+    torsion_potential = potentials.PeriodicTorsion(mol_c_torsion_idxs).bind(mol_c_torsion_params)
+    nonbonded_potential = potentials.NonbondedPairList(
+        mol_c_nbpl_idxs,
+        mol_a_nbpl.get_rescale_mask(),
+        mol_a_nbpl.get_beta(),
+        mol_a_nbpl.get_cutoff(),
+    ).bind(mol_c_nbpl_params)
+
+    return system.VacuumSystem(bond_potential, angle_potential, torsion_potential, nonbonded_potential)
+
+
+def find_dummy_groups_and_anchors(mol_a, mol_b, core_a, core_b):
```

Comment:
Question: do `core_a` and `core_b` need to be the same length? If so, should we check for this explicitly?

Reply:
yes - will add an assertion, they also need to be unique as well